include/nova_renderer/device_memory_resource.hpp

```diff
@@ -3,16 +3,16 @@
 #include <foundational/allocation/allocation_structs.hpp>
 #include <foundational/allocation/bytes.hpp>
 #include <foundational/allocation/size_only_allocator.hpp>
+#include "rhi_types.hpp"
 
 using namespace foundational::allocation::operators;
 
 namespace nova::renderer {
-    namespace rhi {
-        struct DeviceMemory;
-    }
 
     struct DeviceMemoryAllocation {
-        rhi::DeviceMemory* memory = nullptr;
+        //rhi::DeviceMemory* memory = nullptr;
+        rhi::DeviceMemory memory{}; // I suppose should be it owner!
```

Comment:
This won't work... Like at all

First off, Nova will have a lot of DeviceMemoryAllocations for a single DeviceMemory. The DeviceMemoryAllocation doesn't own the DeviceMemory

Even if it did own the DeviceMemory it was allocated from, DeviceMemory is a struct that API-specific structs inherit from. Nova uses pointers for all the RHI structs to allow each RHI backend to store its own data for an image, descriptor set, etc. We shouldn't use value types anywhere for RHI structs